File: wintest/steps/hotkey.py
"""Hotkey step -- press a key combination."""

from ._base import StepDefinition, FieldDef
from ..tasks.schema import StepResult


def validate(step, step_num):
    if not step.keys or len(step.keys) < 2:
        return [f"Step {step_num}: 'hotkey' requires a 'keys' list with at least 2 keys"]
    return []


def execute(step, agent):
    agent.actions.hotkey(*step.keys)
    return StepResult(step=step, passed=True)


definition = StepDefinition(
    name="hotkey",
    description="Press a key combination (ctrl+c, alt+f4, ...)",
    fields=[FieldDef("keys", "string[]", required=True)],
    validate=validate,
    execute=execute,
)

File: wintest/ui/web/frontend/src/docs/hotkey.ts
import type { StepDoc } from './types';

export const hotkeyDoc: StepDoc = {
  name: 'hotkey',
  title: 'Hotkey',
  summary: 'Press a key combination.',
  description:
    'Presses multiple keys simultaneously (a keyboard shortcut). The keys are held down in order and released in reverse order. Requires at least two keys. Common modifier keys: ctrl, alt, shift, win.',
  parameters: [
    {
      name: 'keys',
      type: 'string[]',
      required: true,
      description:
        'List of keys to press together (minimum 2). Examples: ["ctrl", "c"] for copy, ["ctrl", "shift", "s"] for save-as, ["alt", "f4"] to close a window.',
    },
  ],
  example:
    '- type: hotkey\n  keys: ["ctrl", "s"]\n  description: "Save the file"',
};
